### PR TITLE
Rework resizers

### LINKED ISSUE #791: Panel resizing improvements
- [x] Use Figma design for the styling
  - [x] small icon
  - [x] icon only visible when hovered
  - [x] mouse icon changes when hovering: `SetCursor` is not implemented on some platforms.
- [x] only one icon per axis, depending on the orientation
- [x] Rework panel creation code to take into account linked panels. Currently, the panels are always split in half. (https://github.com/Nexus-Mods/NexusMods.App/pull/813)
- [x] Rework tests. (https://github.com/Nexus-Mods/NexusMods.App/pull/813)
- [x] Figure out what to do when the orientation changes from vertical to horizontal and vise versa.

### PR DESCRIPTION
**Pull request by erri120**
Resolves #791

This PR rewrites the resizer code to make it more robust and to follow the Figma design.

### CODE REVIEW

File: src/NexusMods.App.UI/WorkspaceSystem/PanelResizer/PanelResizerView.axaml.cs [hunk 1]
```diff
@@ -61,30 +81,25 @@ public PanelResizerView()
                 .Do(_ =>
                 {
                     _isPressed = false;
-                    _startPoint = new Point(0, 0);
+                    Icon.IsVisible = false;
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/PanelResizer/PanelResizerView.axaml.cs:84]: Don't set this to invisible here, it should only become invisible when the pointer exits.

File: src/NexusMods.App.UI/WorkspaceSystem/PanelResizer/PanelResizerView.axaml.cs [hunk 2]
```diff
@@ -44,12 +64,12 @@ public PanelResizerView()
...
                 {
                     _isPressed = true;
-                    _startPoint = ViewModel!.ActualPosition;
+                    Icon.IsVisible = true;
                 })
                 .Finally(() =>
                 {
                     _isPressed = false;
-                    _startPoint = new Point(0, 0);
+                    Icon.IsVisible = false;
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/PanelResizer/PanelResizerView.axaml.cs:72]: Don't set invisible here, should be handled by pointer exit.

File: src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs [hunk 3]
```diff
@@ -438,94 +438,245 @@ internal static bool IsPerfectGrid(WorkspaceGridState gridState)
...
+
+                columnsToAdd[numColumnsToAdd++] = column;
+            }
 
-            // Step 1.1: go through all panels and find their adjacent panels
-            foreach (var other in currentState)
+            if (minX > maxX)
             {
-                if (other.Key == current.Key) continue;
+                Debugger.Break();
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs:498]: Do you want to leave this in, or is it a leftover from debugging?

File: src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs [hunk 4]
```diff
@@ -438,94 +438,245 @@ internal static bool IsPerfectGrid(WorkspaceGridState gridState)
...
+            var column = columnEnumerator.Current;
+            var info = column.Info;
+            var rows = column.Rows;
+
+            for (var i = 0; i < rows.Length - 1; i++)
+            {
+                var (curId, curRect) = rows[i];
+                var (nextId, nextRect) = rows[i + 1];
+
+                // if (!curRect.X.IsCloseTo(nextRect.X) || !curRect.Right.IsCloseTo(nextRect.Right)) continue;
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs:588]: Should be cleaned up?

File: src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs [hunk 5]
```diff
@@ -438,94 +438,245 @@ internal static bool IsPerfectGrid(WorkspaceGridState gridState)
...
+    internal static List<ResizerInfo> GetResizers2(WorkspaceGridState workspaceState)
     {
-        var tmp = new List<ResizerInfo>(capacity: currentState.Count);
-        var adjacentPanels = new List<ResizerInfo>(capacity: currentState.Count);
+        return workspaceState.IsHorizontal
+            ? GetResizersForHorizontal(workspaceState)
+            : GetResizersForVertical(workspaceState);
+    }
+
+    private static List<ResizerInfo> GetResizersForVertical(WorkspaceGridState currentState)
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs:448]: Could you add some comments here and there, to make the code easier to read. 
These are long functions and following what is going on gets hard.

File: tests/NexusMods.UI.Tests/WorkspaceSystem/GridUtilsTests/GetResizersTests.cs [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+[SuppressMessage("ReSharper", "HeapView.BoxingAllocation")]
+[SuppressMessage("ReSharper", "HeapView.ObjectAllocation.Evident")]
+[SuppressMessage("ReSharper", "HeapView.ObjectAllocation")]
+public partial class GridUtilsTests
+{
+    [Theory]
+    [MemberData(nameof(TestData_GetResizers2_Generated))]
+    [SuppressMessage("Usage", "xUnit1026:Theory methods should use all of their parameters")]
+    public void Test_GetResizers2(
```
Comment (Al12rs) [inline tests/NexusMods.UI.Tests/WorkspaceSystem/GridUtilsTests/GetResizersTests.cs:16]: probably needs to be renamed without the 2

**Review by Al12rs — CHANGES_REQUESTED** [review]

File: src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs [hunk 7]
```diff
@@ -438,94 +438,245 @@ internal static bool IsPerfectGrid(WorkspaceGridState gridState)
         return res.UnionById(updates);
     }
 
-    internal static IReadOnlyList<ResizerInfo> GetResizers(
-        ImmutableDictionary<PanelId, Rect> currentState,
-        bool isWorkspaceHorizontal = true)
+    internal static List<ResizerInfo> GetResizers2(WorkspaceGridState workspaceState)
```
[the remark this answers is not in the document]
Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs:441]: That was just a leftover, while working on the rework I kept the old `GetResizers` and created a new method called `GetResizers2`. I'll update the name.

File: src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs (hunk 3 above)
Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs:498]: I'm not sure. The condition `minX > maxX` should be impossible, but our code depends on that assumption to be correct. I think I'll use a `Debug.Assert` for this.

**Review by Al12rs — APPROVED** [review]

**Pull request merged**